pkg/apis/common/v1/validations.go

```diff
@@ -76,6 +76,20 @@ func CheckSupportedStackVersion(ver string, supported version.MinMaxVersion) fie
 	return nil
 }
 
+// CheckDeprecatedStackVersion checks that the given version is not deprecated.
+func CheckDeprecatedStackVersion(ver string, deprecatedVersion version.MinMaxVersion) (string, field.ErrorList) {
```

Comment:
It seems that the second parameter is always `version.DeprecatedVersions`?
```suggestion
func CheckDeprecatedStackVersion(ver string) (string, field.ErrorList) {
```